src/api/find2_0_commons.cpp

```diff
@@ -38,12 +38,11 @@
 
 #include <nlohmann/json.hpp>
 
-extern "C" {
-miopenStatus_t miopenCreateConvProblem(miopenProblem_t* problem,
-                                       miopenConvolutionDescriptor_t operatorDesc,
-                                       miopenProblemDirection_t direction)
+template <class OperationDescriptor>
+miopenStatus_t MakeProblem(miopenProblem_t* problem,
```

Comment:
```suggestion
static miopenStatus_t MakeProblem(miopenProblem_t* problem,
```

Reply:
It is template and therefor inline, so static and empty namespace won't affect it.

Reply:
https://en.cppreference.com/w/cpp/language/storage_duration
> Any of the following names declared at namespace scope have internal linkage ... functions, or function templates declared static ...
Any of the following names declared at namespace scope have external linkage, unless they are declared in an unnamed namespace ... names of all templates not listed above (that is, not function templates declared static) ...